src/riscv.h

```diff
@@ -227,6 +227,10 @@ state_t *state_new(void);
 /* delete a state */
 void state_delete(state_t *s);
 
+#if RV32_HAS(JIT)
+void rv_profile(riscv_t *rv);
```

Comment:
The function `rv_profile` needs to be designed as independent of the JIT configuration. This means it should be callable in both interpreter-only mode and when using the tier-1 JIT compiler, enabling it to gather runtime profiling data for subsequent analysis. Additionally, we should provide tools capable of handling graphical representations of this data.